internal/provider/generate.go

```diff
@@ -414,7 +414,7 @@ func (g *generator) renderStaticWebsite(providerName string, providerSchema *tfj
 		g.infof("rendering %q", rel)
 		switch relDir {
 		case "data-sources/":
-			resName := shortName + "_" + removeAllExt(relFile)
+			resName := resourceName(shortName, relFile)
```

Comment:
Question: isn't `shortName` the Provider shortname here?

In the case of the HTTP provider (through which you identified the issue), I believe the `http` refers to the resource. Otherwise if, for example, you decide to create a second resource in this provider (say, for the sake of it, `https`), you won't be able to match it OR it would match the wrong file.

I'll leave a comment in the unit test below.